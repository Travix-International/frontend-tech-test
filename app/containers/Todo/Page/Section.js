import styled from 'styled-components';

const Section = styled.section`
  margin: 1em auto;

  &:first-child {
    margin-top: 0;
  }
`;

export default Section;
