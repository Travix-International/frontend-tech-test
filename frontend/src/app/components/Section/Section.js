import React from 'react';
import PropTypes from 'prop-types';
import Wrapper from './assets/style';

/**
 * Shows white background div with padding
 *
 * @param {node} children - Everything in between of the button opening and closing tag (like Component)
 */
const Section = ({ children }) => {
  return <Wrapper>{children}</Wrapper>;
};

Section.propTypes = {
  children: PropTypes.node.isRequired
};

export default Section;
