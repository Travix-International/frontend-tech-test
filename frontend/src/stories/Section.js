import React from 'react';
import { storiesOf } from '@storybook/react';
import Section from '../app/components/Section';

storiesOf('Section', module).add('default', () => (
  <Section>Nothing special. It is just a simple section with some paddings.</Section>
));
