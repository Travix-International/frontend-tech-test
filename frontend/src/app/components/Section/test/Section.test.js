import React from 'react';
import { shallow } from 'enzyme';
import Section from '../Section';

it('Section renders correctly', () => {
  const SectionSnapshot = shallow(<Section> Some section</Section>);
  expect(SectionSnapshot).toMatchSnapshot();
});
