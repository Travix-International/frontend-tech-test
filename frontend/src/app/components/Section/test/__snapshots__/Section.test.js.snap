// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders correctly 1`] = `ShallowWrapper {}`;
